myokit/formats/sbml/_writer.py

```diff
@@ -0,0 +1,339 @@
+        tree.write(
+            path,
+            encoding='utf-8',
+            method='xml',
+            xml_declaration=True,
+            pretty_print=True,
+        )
+
+    def write_string(model: Model) -> str:
+        tree = SBMLWriter._model(model)
+        return etree.tostring(
+            tree,
+            encoding='utf-8',
+            method='xml',
+            pretty_print=True,
+        )
+
+    def _compartment(
+            compartment: Compartment,
+            unit_to_str: dict
+    ) -> etree.Element:
+        node = etree.Element('compartment', id=compartment.sid())
+        if compartment.size_units() != myokit.units.dimensionless:
+            node.attrib['units'] = unit_to_str[compartment.size_units()]
+        if compartment.spatial_dimensions() is not None:
+            node.attrib['spatialDimensions'] = str(
+                compartment.spatial_dimensions()
+            )
+        return node
+
+    def _unit(sid: str, unit: myokit.Unit) -> etree.Element:
+        node = etree.Element('unitDefinition', id=sid)
+        kinds = [
+            'gram', 'metre', 'second',
+            'ampere', 'kelvin', 'candela', 'mole'
+        ]
+        multiplier = unit.multiplier()
+        for exponent, kind in zip(unit.exponents(), kinds):
+            if exponent != 0:
+                child = etree.Element('unit')
+                child.attrib['kind'] = kind
+                child.attrib['exponent'] = str(exponent)
+                if multiplier is not None:
+                    child.attrib['multiplier'] = str(multiplier)
+                    multiplier = None
+                node.append(child)
+        # might also have a dimensionless unit and a multiplier
+        if multiplier is not None:
+            child = etree.Element('unit')
+            child.attrib['kind'] = 'dimensionless'
+            child.attrib['multiplier'] = str(multiplier)
+            node.append(child)
+        return node
+
+    def _parameter(
+            parameter: Parameter,
+            unit_to_str_map: dict
+    ) -> Tuple[etree.Element, etree.Element, etree.Element]:
+        """
+        returns the XML representation of this parameter as a tuple of
+        (parameter, initial_assignment, rule).
+        """
+        parameter_xml = etree.Element('parameter', id=parameter.sid())
+        if (
+            parameter.units() is not None and
+            parameter.units() != myokit.units.dimensionless
+        ):
+            parameter_xml.attrib['units'] = unit_to_str_map[parameter.units()]
+
+        if parameter.is_constant():
+            parameter_xml.attrib['constant'] = 'true'
+            value = parameter.value().eval()
+            parameter_xml.attrib['value'] = str(value)
+            return parameter_xml, None, None
+        else:
+            initial_assignment, rule = SBMLWriter._quantity(parameter)
+            return parameter_xml, initial_assignment, rule
+
+    def _math(expression: myokit.Expression) -> etree.Element:
```

Comment:
I'm guessing this is all pretty fast, so there's no point making this class a single-use object that can reuse the same mathmlexpressionwriter ?